app/lib/handlers.dart

```diff
@@ -62,6 +62,8 @@ Future<shelf.Response> appHandler(
     return docHandler(request);
   } else if (path.startsWith('/static')) {
     return staticsHandler(request);
+  } else if (path.startsWith('/flutter')) {
+    return flutterHandler(request);
```

Comment:
I think you should move the handling of this to be precisely like we do with '/packages', namely add it to the `handlers` map above, then it gets called only if the url path is correct (see comment below why)

You can add an extra redirection handler here for `/flutter` -> `/flutter/plugins`